hackathon_site/hardware/views.py

```diff
@@ -62,9 +60,32 @@ class IncidentListView(mixins.ListModelMixin, generics.GenericAPIView):
     filter_backends = (filters.DjangoFilterBackend, SearchFilter)
     filterset_class = IncidentFilter
 
+    def get_queryset(self):
+        queryset = Incident.objects.all()
+        if self.request.method == "GET":
+            return queryset.select_related(
```

Comment:
You can just return this in all cases (and just keep it as the `queryset` property on the view, don't need the `get_queryset` method).

Since your serializer includes a method field for getting the team id, it would be efficient to at least include `order_item__order__team` in both cases. Technically the related hardware isn't required for the post case, so if you want to keep the `get_queryset` method you can `select_related` team accordingly in the POST block

Reply:
ok the get_queryset method is removed.